### ISSUE TITLE
incorrectly removes `{` in certain `use` expressions

### ISSUE THREAD

**Issue by the-shank** [dtolnay/prettyplease#69]
Minimum reproducible example:

```rust
const INPUT: &str = stringify!{
use ring::signature::{foo as ringsig};
};

fn main() {
        let syntax_tree = syn::parse_file(INPUT).unwrap();
        let formatted = prettyplease::unparse(&syntax_tree);
        print!("{}", formatted);
}
```

The output is
```
❯ cargo run --quiet
use ring::signature::foo as ringsig;
```

Notice the missing `{` and `}` around `foo as ringsig`.

**Comment by dtolnay** [comment]
```console
$ echo 'use ring::signature::{foo as ringsig};' | rustfmt
use ring::signature::foo as ringsig;
```

**Comment by the-shank** [comment]
@dtolnay I am super confused now:

Indeed:
```
❯ echo 'use ring::signature::{foo as ringsig};' | rustfmt
use ring::signature::foo as ringsig;
```

But:
```
❯ echo 'use Foo::Bar::{self as foobar};' | rustfmt
use Foo::Bar::{self as foobar};
```

I have no idea as to why the braces remain in the second example ??

**Comment by dtolnay** [comment]
`use foo::bar::self as foobar;` is not legal syntax in Rust, and `use foo::bar as foobar;` means a different thing than `use foo::bar::{self as foobar};`.

**Comment by the-shank** [comment]
Yes, I agree. But prettyplease removes the braces in the `{self as xxx}` case as well.

```rust
const INPUT: &str = stringify! {
use ring::signature::{foo as ringsig};
use Foo::Bar::{self as foobar};
};

fn main() {
    let syntax_tree = syn::parse_file(INPUT).unwrap();
    let formatted = prettyplease::unparse(&syntax_tree);
    print!("{}", formatted);
}
```

Output:
```
❯ cargo run --quiet
use ring::signature::foo as ringsig;
use Foo::Bar::self as foobar;
```

**Comment by dtolnay** [comment]
I've published a fix in 0.2.17 to keep braces around `{self as …}` while continuing to discard them in other cases.
